cargo-pgrx/src/command/upgrade.rs

```diff
@@ -0,0 +1,216 @@
+//LICENSE Portions Copyright 2019-2021 ZomboDB, LLC.
+//LICENSE
+//LICENSE Portions Copyright 2021-2023 Technology Concepts & Design, Inc.
+//LICENSE
+//LICENSE Portions Copyright 2023-2023 PgCentral Foundation, Inc. <contact@pgcentral.org>
+//LICENSE
+//LICENSE All rights reserved.
+//LICENSE
+//LICENSE Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+use cargo_edit::{registry_url, Dependency};
+use eyre::eyre;
+use std::path::{Path, PathBuf};
+use toml_edit::KeyMut;
+use tracing::{debug, error, info, warn};
+
+use crate::CommandExecute;
+
+/// Upgrade pgrx crate versions in `Cargo.toml`.
+/// Defaults to latest.
+#[derive(clap::Args, Debug)]
+#[clap(author)]
+pub(crate) struct Upgrade {
+    /// Specify a version to upgrade to, rather than defaulting to the latest
+    /// version.
+    #[clap(long, short)]
+    pub(crate) target_version: Option<String>, //TODO: typed version not string
+
+    /// Path to the manifest file (usually Cargo.toml). Defaults to
+    /// "./Cargo.toml" in the working directory.
+    #[clap(long, short)]
+    pub(crate) path: Option<PathBuf>,
+
+    // Flag to permit pre-release builds
+    #[clap(long, short)]
+    pub(crate) allow_prerelease: bool,
+}
+trait DependencySource {
+    fn set_version<A: AsRef<str>>(&mut self, new_version: A);
+    fn get_version(&self) -> Option<&String>;
+}
+impl DependencySource for cargo_edit::Source {
```

Comment:
nit: this is the only implementation for the trait, and you only use it once below (in `replace_version`). i assume it's a trait because you can't add inherent functions to `Source` since it's in another crate. i kinda would prefer to make `set_version` and `get_version` free functions instead to make it clear it's not generic over the source.